code/game/machinery/atmos_exporter.dm

```diff
@@ -0,0 +1,293 @@
+					concentration += p_concentration3
+					price = p_price
+			updateUsrDialog()
+		if ("scan")
+			for(var/obj/machinery/portable_atmospherics/canister/C in get_turf(can_turf))
+				atmosanalyzer_scan(C.air_contents, usr, C)
+				playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 100, 1)
+		if ("send")
+			for(var/obj/machinery/portable_atmospherics/canister/C in get_turf(can_turf))
+				if((lpressure) <= C.air_contents.return_pressure() && C.air_contents.return_pressure() <= (upressure))
+					if((ltemp) <= C.air_contents.temperature && C.air_contents.temperature <= (utemp))
+						var/total_moles = C.air_contents.total_moles()
+						var/list/can_gas = C.air_contents.gases
+						var/i = 0
+						for(var/id in gas)
+							if(can_gas[id])
+								var/gas_concentration = (can_gas[id][MOLES]/total_moles)*100
+								i++
+								if(gas_concentration <= (concentration[i]-5) || gas_concentration >= (concentration[i]+5))
+									visible_message("<span class='danger'>Invalid gas composition detected!</span>")
+									playsound(src, 'sound/machines/defib_failed.ogg', 100, 1)
+									return
+							else
+								visible_message("<span class='danger'>Invalid gas detected!</span>")
+								playsound(src, 'sound/machines/defib_failed.ogg', 100, 1)
+								return
+						visible_message("<span class='danger'>Valid [export_target] canister detected... exporting now. Your department will receive [price] points!</span>")
+						C.anchored = 1
+						for(var/obj/machinery/atmos_points_exporter/EX in atmos_exporter_list)
+							EX.ship()
+						for(var/obj/machinery/engi_points_manager/EPM in engi_points_list)
+							EPM.GBP += price
+							EPM.GBPearned += price
+						if(export_target == "premium")
+							visible_message("<span class='danger'>You have been blessed by the atmos gods for exporting a premium canister!</span>")
+							var/prize = pick(/obj/item/clothing/under/rank/atmos_elite,/obj/item/clothing/head/atmos_hood,/obj/item/clothing/neck/cloak/atmos)
+							spawn_atom_to_turf(/obj/effect/overlay/temp/explosion/fast, src, 1, admin_spawn=FALSE)
+							spawn_atom_to_turf(prize, src, 1, admin_spawn=FALSE)
+							playsound(src, 'sound/effects/pray_chaplain.ogg', 100, 1)
+						quality.Remove(export_target)
+						reset()
+					else
+						visible_message("<span class='danger'>Invalid temperature detected!</span>")
+						playsound(src, 'sound/machines/defib_failed.ogg', 100, 1)
+				else
+					visible_message("<span class='danger'>Invalid pressure detected!</span>")
+					playsound(src, 'sound/machines/defib_failed.ogg', 100, 1)
+
+			updateUsrDialog()
+
+/obj/machinery/atmos_points/proc/reset()
+	export_name = "------"
+	export_target = ""
+	gas.Cut()
+	concentration.Cut()
+	upressure = 0
+	lpressure = 0
+	utemp = 0
+	ltemp = 0
+
+/obj/machinery/atmos_points/process()
+	if(market_cooldown < world.time)
+		reset()
+		economy_name = corp_name()
+		e_gas.Cut()
+		var/list/gas_list = list("o2","n2","n2o","plasma","co2")

```

Comment:
>Good one.

Can you not?
And you most likely encountered the list that's used to preserve old, previously hard-coded behaviour, hence why it only contains the pre-refactor gases (eg: no bz, no freon, etc.)

If there's no other gas list, then this is as fine as possible without actually adding a new list, and a new list wouldn't have the value or trust of one that was actually used in the system.